pallets/hyperdrive/src/lib.rs

```diff
@@ -219,32 +241,40 @@ pub mod pallet {
                     && current_block < activity_window.end_block,
                 Error::<T, I>::SubmitOutsideTransmitterActivityWindow
             );
-            ensure!(
-                block
-                    .checked_rem(&T::TransmissionRate::get())
-                    .ok_or(Error::<T, I>::CalculationOverflow)?
-                    .is_zero(),
-                Error::<T, I>::SubmitOutsideTransmissionRate
-            );
 
             // insert merkle root proposal since all checks passed
             // allows for constant-time validity checks
-            let accepted =
-                StateMerkleRootCount::<T, I>::mutate(&block, &state_merkle_root, |count| {
-                    let count_ = count.unwrap_or(0) + 1;
-                    *count = Some(count_);
-                    count_ >= <T as Config<I>>::TransmissionQuorum::get()
-                });
+            let accepted = StateMerkleRootCount::<T, I>::mutate(
+                &snapshot,
+                &state_merkle_root,
+                |submissions| {
+                    // This can be improved once [let chains feature](https://github.com/rust-lang/rust/issues/53667) lands
+                    if let Some(transmitters) = submissions {
+                        if !transmitters.contains(&who) {
```

Comment:
(**) this is linear in the amount of transmitters, while it could be a constant lookup. Should we rather use a separate storage map just to validate who has already submitted?

Reply:
I updated the code to use a `BTreeSet` instead of a vec.

The number of transmitters will be small, less than 20. The old code should also be fine.